test/e2e/serviceloadbalancers.go

```diff
@@ -0,0 +1,292 @@
+	if err != nil {
+		return err
+	}
+
+	// Find the external addresses of the nodes the pods are running on.
+	for _, p := range pods.Items {
+		wait.Poll(pollInterval, serviceRespondingTimeout, func() (bool, error) {
+			address, err := getHostExternalAddress(h.client, &p)
+			if err != nil {
+				Logf("%v", err)
+				return false, nil
+			}
+			h.address = append(h.address, address)
+			return true, nil
+		})
+	}
+	if len(h.address) == 0 {
+		return fmt.Errorf("No external ips found for loadbalancer %v", h.getName())
+	}
+	return nil
+}
+
+func (h *haproxyControllerTester) stop() error {
+	return h.client.ReplicationControllers(h.rcNamespace).Delete(h.rcName)
+}
+
+func (h *haproxyControllerTester) lookup(ingressKey string) string {
+	// The address of a service is the address of the lb/servicename, currently.
+	return fmt.Sprintf("http://%v/%v", h.address[0], ingressKey)
+}
+
+// ingManager starts an rc and the associated service.
+type ingManager struct {
+	rcCfgPaths  []string
+	svcCfgPaths []string
+	ingCfgPath  string
+	name        string
+	namespace   string
+	client      *client.Client
+	svcNames    []string
+}
+
+func (s *ingManager) getName() string {
+	return s.name
+}
+
+func (s *ingManager) start(namespace string) (err error) {
+	// Create rcs
+	for _, rcPath := range s.rcCfgPaths {
+		rc := rcFromManifest(rcPath)
+		rc.Namespace = namespace
+		rc.Spec.Template.Labels["rcName"] = rc.Name
+		rc, err = s.client.ReplicationControllers(rc.Namespace).Create(rc)
+		if err != nil {
+			return
+		}
+		if err = waitForRCPodsRunning(s.client, rc.Namespace, rc.Name); err != nil {
+			return
+		}
+	}
+	// Create services.
+	// Note that it's upto the caller to make sure the service actually matches
+	// the pods of the rc.
+	for _, svcPath := range s.svcCfgPaths {
+		svc := svcFromManifest(svcPath)
+		svc.Namespace = namespace
+		svc, err = s.client.Services(svc.Namespace).Create(svc)
+		if err != nil {
+			return
+		}
+		// TODO: This is short term till we have an Ingress.
+		s.svcNames = append(s.svcNames, svc.Name)
+	}
+	s.name = s.svcNames[0]
+	s.namespace = namespace
+	return nil
+}
+
+func (s *ingManager) test(path string) error {
+	url := fmt.Sprintf("%v/hostName", path)
+	httpClient := &http.Client{}
+	return wait.Poll(pollInterval, serviceRespondingTimeout, func() (bool, error) {
+		body, err := simpleGET(httpClient, url)
+		if err != nil {
+			Logf("%v\n%v\n%v", url, body, err)
+			return false, nil
+		}
+		return true, nil
+	})
+}
+
+var _ = Describe("ServiceLoadBalancer", func() {
+	// These variables are initialized after framework's beforeEach.
+	var ns string
+	var repoRoot string
+	var client *client.Client
+
+	framework := Framework{BaseName: "servicelb"}
+
+	BeforeEach(func() {
+		framework.beforeEach()
+		client = framework.Client
+		ns = framework.Namespace.Name
+		repoRoot = testContext.RepoRoot
+	})
+
+	AfterEach(func() {
+		framework.afterEach()
+	})
+
+	It("Should support simple GET on Ingress ips", func() {
```

Comment:
nit: 'S' should be lower case please, so as to form a correct sentence when the Describe() clause is prefixed on to this string.  Wierd, I know, but that's Ginkgo/Behavior Driven Development.  I don't make the weird rules, I'm just trying to keep everything work approximately sanely :-)